neuralprophet/utils.py

```diff
@@ -987,3 +987,153 @@ def configure_trainer(
+            inputs["events"]["multiplicative"] = sliced_tensor[
+                :, future_offset : future_offset + n_forecasts + n_lags, events_start_idx : events_end_idx + 1
+            ]
+
+        # Unpack additive regressor features
+        if "additive_regressors" in feature_indices:
+            regressors_start_idx, regressors_end_idx = feature_indices["additive_regressors"]
+            future_offset = max_lags - n_lags
+            inputs["regressors"] = OrderedDict()
+            inputs["regressors"]["additive"] = sliced_tensor[
+                :,
+                future_offset : future_offset + n_forecasts + n_lags,
+                regressors_start_idx : regressors_end_idx + 1,
+            ]
+
+        # Unpack multiplicative regressor features
+        if "multiplicative_regressors" in feature_indices:
+            regressors_start_idx, regressors_end_idx = feature_indices["multiplicative_regressors"]
+            future_offset = max_lags - n_lags
+            if "regressors" not in inputs:
+                inputs["regressors"] = OrderedDict()
+            inputs["regressors"]["multiplicative"] = sliced_tensor[
+                :,
+                future_offset : future_offset + n_forecasts + n_lags,
+                regressors_start_idx : regressors_end_idx + 1,
+            ]
+
+        # Unpack seasonality feature
+        if config_seasonality is not None and hasattr(config_seasonality, "periods"):
+            inputs["seasonalities"] = OrderedDict()
+            for seasonality_name in config_seasonality.periods.keys():
+                seasonality_key = f"seasonality_{seasonality_name}"
+                if seasonality_key in feature_indices:
+                    seasonality_start_idx, seasonality_end_idx = feature_indices[seasonality_key]
+                    seasonality_offset = max_lags - n_lags
+                    inputs["seasonalities"][seasonality_name] = sliced_tensor[
+                        :,
+                        seasonality_offset : seasonality_offset + n_forecasts + n_lags,
+                        seasonality_start_idx:seasonality_end_idx,
+                    ]
+
+        # Unpack lagged regressor features
+        if config_lagged_regressors:
+            inputs["covariates"] = OrderedDict()
+            for name, lagged_regressor in config_lagged_regressors.items():
+                lagged_regressor_key = f"lagged_regressor_{name}"
+                if lagged_regressor_key in feature_indices:
+                    lagged_regressor_start_idx, _ = feature_indices[lagged_regressor_key]
+                    covar_lags = lagged_regressor.n_lags
+                    lagged_regressor_offset = max_lags - covar_lags
+                    inputs["covariates"][name] = sliced_tensor[
+                        :,
+                        lagged_regressor_offset : lagged_regressor_offset + covar_lags,
+                        lagged_regressor_start_idx,
+                    ]
+
+    else:
```

Comment:
Please add a comment explaining dimensions and why most have a `unsqueeze(1)` added.